compiler/rustc_borrowck/src/diagnostics/move_errors.rs

```diff
@@ -414,11 +414,23 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
             && use_spans.map_or(true, |v| !v.for_closure())
             && !has_complex_bindings
         {
-            err.span_suggestion_verbose(
+            let span = match use_spans {
+                Some(UseSpans::FnSelfUse { var_span, fn_call_span, .. }) => {
+                    // `fn_call_span` is the span of `unwrap()`
+                    // This gets the 'variable' span excluding the unwrap
+                    let span = var_span.until(fn_call_span);
+
+                    // Unfortunately the new span has a "." on the end which makes
+                    // the suggestion format strangely, so we adjust the hi bound manually
+                    span.with_hi(span.hi() - BytePos(1))
```

Comment:
If it has `.` on the end, then maybe we should suggest `as_ref().` instead of `.as_ref()`.

Span arithmetic is dangerous because macros can set spans arbitrarily and `span - BytePos(1)` can potentially underflow or point into the middle of a unicode character producing an ICE.

Reply:
Sounds good to me, I'll remove the span arithmetic. This + the previous suggestion gives this output for `issue-96438.rs`:

```rs
error[E0507]: cannot move out of dereference of `RefMut<'_, Option<Vec<()>>>`
 --> ./src/test/ui/borrowck/issue-96438.rs:4:5
  |
4 |     cell.borrow_mut().unwrap().pop().unwrap();
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ move occurs because value has type `Option<Vec<()>>`, which does not implement the `Copy` trait
  |
help: consider borrowing the `Option`'s content
  |
4 |     cell.borrow_mut().as_ref().unwrap().pop().unwrap();
  |                       +++++++++
help: consider borrowing the `Option`'s content
  |
4 |     cell.borrow_mut().as_mut().unwrap().pop().unwrap();
  |                       +++++++++

error: aborting due to previous error

For more information about this error, try `rustc --explain E0507`.
```

Which is alright, however it doesn't seem to fix the double application in `option-content-move.stderr` unfortunately.